pkgs/applications/virtualization/arion/default.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs
+, haskellPackages ? pkgs.haskellPackages
+}:
+let
+  optionalCabal2nixFile = ./cabal2nix-arion-compose.nix;
+
+  arion-compose =
+    # Test suite needs nixpkgs as dependency which would make it quite heavy.
+    pkgs.haskell.lib.dontCheck (
+      if ! builtins.pathExists optionalCabal2nixFile
+      then haskellPackages.arion-compose
+      else haskellPackages.callPackage optionalCabal2nixFile {});
```

Comment:
Instead of hard-coding the path `./cabal2nix-arion-compose.nix`, why not take `optionalCabal2nixFile` as an argument to this file, so you'd have something like:

```nix
{ pkgs
, haskellPackages ? pkgs.haskellPackages
, optionalCabal2nixFile ? null
}:
```

Then, the `arion-compose` expression would become:

```nix
  arion-compose =
    # Test suite needs nixpkgs as dependency which would make it quite heavy.
    if isNull optionalCabal2nixFile then haskellPackages.arion-compose else pkgs.haskell.lib.dontCheck (haskellPackages.callPackage optionalCabal2nixFile {});
```

This should make it more natural to use for most users who are used to the `callPackage` style of packages.